src/app/charts/chart.component.ts

```diff
@@ -137,6 +158,11 @@ export class ChartComponent implements OnChanges {
         this.imageExportService.downloadAsPNG(this.chart.indicator.name, fileName);
     }
 
+    public onThresholdSelected($event) {
+        delete $event.event; // everything else is a threshold parameter
```

Comment:
This comment seems like something that could be explicitly addressed by TypeScript by defining an interface for the event emitter that sends this object, rather than having to implicitly 'know' that everything other than `$event.event` is what we want, which is both opaque and an easy way to introduce bugs when refactoring or adding features later.